### ISSUE TITLE
Index(SparseArray([int])) has object type

### ISSUE THREAD

**Issue by TomAugspurger** [pandas-dev/pandas#26972]
```python
In [2]: pd.Index(pd.SparseArray([1, 2]))
Out[2]: Index([1, 2], dtype='object')
```

Should this be an Int64Index?

**Comment by TomAugspurger** [comment]
This also affects IntegerArray

```python
In [9]: pd.Index(pd.array([1, 2], dtype='Int64'))
Out[9]: Index([1, 2], dtype='object')
```

in that case, object-dtype may be more appropriate since NumPy can't represent nullable int64 data.

**Comment by jbrockmendel** [comment]
closed by #49307
